backward.hpp

```diff
@@ -1673,19 +1674,27 @@ namespace Color {
 
 class Colorize {
 public:
-	Colorize(std::FILE* os):
-		_os(os), _reset(false), _istty(false) {}
+	Colorize(std::ostream& os):
+		_os(os), _reset(false), _use_colors(false) {}
 
-	void init() {
-		_istty = isatty(fileno(_os));
+	void activate() {
+		_use_colors = true;
+		// in a colorful environment, reset at the beginning
+		set_color(Color::reset);
+	}
+
+	void activate_if_tty(std::FILE *desc) {
+		if (isatty(fileno(desc))) {
+			activate();
+		}
 	}
 
 	void set_color(Color::type ccode) {
```

Comment:
let's have two version of set_color, one that takes an ostream, the other a FILE*.